app/web/src/components/Navigation/Navigation.tsx

```diff
@@ -127,14 +172,60 @@ const useStyles = makeStyles((theme) => ({
     alignSelf: "center",
     fontWeight: "bold",
   },
+  menuContainer: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+  },
+  menu: {
+    boxShadow: theme.shadows[1],
+    minWidth: "12rem",
+  },
+  menuPopover: {
+    transform: "translateY(1rem)",
+  },
+  menuItem: {
+    width: "100%",
+  },
+  notificationCount: {
+    color: grey[500],
+    fontWeight: "bold",
+  },
+  menuBtn: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+    border: `1px solid ${grey[300]}`,
+    borderRadius: 999,
+    backgroundColor: grey[200],
+    padding: theme.spacing(1),
+    transition: `${theme.transitions.duration.short}ms ${theme.transitions.easing.easeInOut}`,
+    "&:hover": {
+      opacity: 0.8,
+      backgroundColor: grey[300],
+    },
+  },
+  avatar: {
+    height: "2rem",
+    width: "2rem",
+  },
+  badge: {
+    "& .MuiBadge-badge": {
+      right: "-4px",
+      top: "4px",
+    },
+  },
 }));
 
 export default function Navigation() {
   const authClasses = useAuthStyles();
   const classes = useStyles();
   const authenticated = useAuthContext().authState.authenticated;
-  const [open, setOpen] = React.useState(false);
+  const [open, setOpen] = useState(false);
+  const [menuOpen, setMenuOpen] = useState(false);
+  const menuRef = React.useRef<HTMLDivElement>(null);
```

Comment:
```suggestion
  const menuRef = React.useRef<HTMLButtonElement>(null);
```
Because of suggested change further down the file